### PR TITLE
UI-Test Automate: Automount

### PR DESCRIPTION
**Pull request by serg-cymbaluk**
Web UI tests for "Automount" functionality.

Ticket: https://pagure.io/freeipa/issue/7735

### CODE REVIEW

**Comment by mrizwan93** [comment]
LGTM. 
Hi @pvoborni,
could you have a look too?

File: ipatests/test_webui/test_automount.py [hunk 1]
```diff
@@ -420,6 +420,38 @@ def test_add_key_dialog(self):
...
+        self.add_record(LOC_ENTITY, LOC_DATA)
+        self.navigate_to_record(LOC_PKEY)
+
+        self.add_record(LOC_ENTITY, MAP_DATA, facet='maps', navigate=False)
+        self.navigate_to_record(MAP_PKEY)
+
+        for missing_field, key in keys.items():
+            self.add_record(MAP_ENTITY, key, negative=True,
+                            facet='keys', navigate=False)
+            assert self. has_form_error(missing_field)
```
Comment (celestian) [inline ipatests/test_webui/test_automount.py]: There is a space between . and has_form_error?

Comment (serg-cymbaluk) [reply ipatests/test_webui/test_automount.py]: Fixed

**Comment by pvoborni** [comment]
In general, I like the approach. The new concept in using entity abstraction class for creation and deletion seems good. I was thinking whether it needs some comments but it was quite obvious after reading the rest.

Hint you can replace in most cases:
```python
self.ui.navigate_to_entity(LOC_ENTITY)
self.ui.delete_record(self.location['pkey'])
```

with just

```python
self.delete(LOC_ENTITY, self.location)
```

**Comment by serg-cymbaluk** [comment]
> In general, I like the approach. The new concept in using entity abstraction class for creation and deletion seems good. I was thinking whether it needs some comments but it was quite obvious after reading the rest.
> 
> Hint you can replace in most cases:
> 
> ```python
> self.ui.navigate_to_entity(LOC_ENTITY)
> self.ui.delete_record(self.location['pkey'])
> ```
> 
> with just
> 
> ```python
> self.ui.delete(LOC_ENTITY, [self.location])
> ```

@pvoborni thank you for the hint. I've fixed this lines.

**Comment by serg-cymbaluk** [comment]
@celestian @pvoborni 
Any other remarks?

**Comment by celestian** [comment]
Everything done from my POV. Tests are passing. => ACK

**Comment by serg-cymbaluk** [comment]
master:

* 1d4b43efa4d0dc48f3541a89d4d645e866b0648f UI tests for "Automount": check "Add Automount..." dialogs
* e957e0aef0273070c6823df6217a262cf3c0c0f3 UI tests for "Automount": check modifying map and key settings
* 97f158aeec029369588a8bbe0cba6b431e855b5f UI tests for "Automount": Fix item deleting
* de06bf2778fb7974f06a6411357b81752ac2626b UI tests for "Automount": check creating indirect automount map without some fields
* b7a149feb22700ccc3a84bb74799b75df963ec23 UI tests for "Automount": check creating automount key without some fields
* ba40590e9d0bec3846d7b13c543de1652f3203de UI tests for "Automount": check indirect map duplication
* 6444808fd2c4bc408543127aafc1835c349fba29 UI tests for "Automount": check some negative cases
* 8f2a75cb00eeef51712368a4e4f71353d593e054 UI tests for "Automount": check dialog confirmation using ENTER
* 8954521007891b85d159d8b3bfa72da04b5a9279 WebUI tests: Make possible to use kwargs with @screenshot decorator

**Pull request closed without merging**